sdk/identity/Azure.Identity/tests/ManagedIdentityCredentialTests.cs

```diff
@@ -318,7 +318,7 @@ public async Task VerifyMsiUnavailableOnIMDSAggregateExcpetion()
             using var environment = new TestEnvVar(new() { { "MSI_ENDPOINT", null }, { "MSI_SECRET", null }, { "IDENTITY_ENDPOINT", null }, { "IDENTITY_HEADER", null }, { "AZURE_POD_IDENTITY_AUTHORITY_HOST", "http://169.254.169.001/" } });
 
             // setting the delay to 1ms and retry mode to fixed to speed up test
-            var options = new TokenCredentialOptions() { Retry = { Delay = TimeSpan.FromMilliseconds(1), Mode = RetryMode.Fixed } };
+            var options = new TokenCredentialOptions() { Retry = { Delay = TimeSpan.FromMilliseconds(1), Mode = RetryMode.Fixed, NetworkTimeout = TimeSpan.FromMilliseconds(100) } };
```

Comment:
Let's add a test that validates that the timeout only gets set on the first message.

Reply:
I'm still adding tests for this and `DefaultAzureCredential` as well, it's just taking a bit since it requires standing up a listener to force network timeouts etc.